File: src/RoboTasker.Api/Models/Capabilities/UpdateCapabilityRequest.cs
using RoboTasker.Application.Robots.Capabilities.CreateCapability;

namespace RoboTasker.Api.Models.Capabilities;

public class UpdateCapabilityRequest
{
    public string? Name { get; set; }
    public string? Description { get; set; }
    public IList<Guid>? DeletedCapabilities { get; set; }
    public IList<CreateCapabilityItem>? NewCapabilities { get; set; }
}

File: src/RoboTasker.UI/src/app/models/robots/capabilities/requests/update-capability-request.ts
import {Guid} from 'guid-typescript';
import {CreateCapabilityItemRequest} from './create-capability-request';

export interface UpdateCapabilityRequest {
  name?: string | undefined;
  description?: string | undefined;
  deletedCapabilities?: Guid[] | undefined;
  newCapabilities?: CreateCapabilityItemRequest[] | undefined;
}
